### PR TITLE
bump NN to v55.0.0

### PR DESCRIPTION
**Pull request by RingerJK**
### Changelog

```
<changelog>Added new `HistoryEvent`: `HistoryEventPushHistoryRecord` allows read events set with `HistoryRecorderHandle#pushHistory`</changelog>
```

### CODE REVIEW

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/history/model/HistoryEventPushHistoryRecord.kt [hunk 1]
```diff
@@ -0,0 +1,47 @@
+package com.mapbox.navigation.core.history.model
+
+/**
+ * Allows to read events set with [com.mapbox.navigator.HistoryRecorderHandle.pushHistory]
+ *
+ * @param type corresponded event type of [com.mapbox.navigator.HistoryRecorderHandle.pushHistory]
+ * @param properties corresponded eventJson of [com.mapbox.navigator.HistoryRecorderHandle.pushHistory]
+ */
+class HistoryEventPushHistoryRecord internal constructor(
```
Comment (RingerJK) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/history/model/HistoryEventPushHistoryRecord.kt:9]: no tests as we still don't expose `HistoryRecorderHandle#pushHistory `

File: libnavigator/src/main/java/com/mapbox/navigation/navigator/internal/MapboxNativeNavigatorImpl.kt [hunk 2]
```diff
@@ -245,8 +245,8 @@ object MapboxNativeNavigatorImpl : MapboxNativeNavigator {
      */
     override suspend fun getRoute(url: String): Expected<RouterError, String> {
         return suspendCoroutine { continuation ->
-            nativeRouter!!.getRoute(url) {
-                continuation.resume(it)
+            nativeRouter!!.getRoute(url) { expected, _ ->
```
Comment (Guardiola31337) [inline libnavigator/src/main/java/com/mapbox/navigation/navigator/internal/MapboxNativeNavigatorImpl.kt:248]: Noting that `Router` is currently `@Deprecated` 👀 

```java
/** @deprecated Public Router class for onboard router is redundant. Please use: {@link com.mapbox.navigator.RouterFactory} instead. */
```

Also the `_` is the `RouterOrigin`.

File: libnavigation-router/src/test/java/com/mapbox/navigation/route/internal/onboard/MapboxOnboardRouterTest.kt [hunk 3]
```diff
@@ -80,7 +80,7 @@ class MapboxOnboardRouterTest {
         every { isValue } returns false
         every { isError } returns true
         every { value } returns null
-        every { error } returns RouterError(FAILURE_MESSAGE, FAILURE_CODE)
+        every { error } returns RouterError(FAILURE_MESSAGE, FAILURE_CODE, REQUEST_CODE)
```
Comment (Guardiola31337) [inline libnavigation-router/src/test/java/com/mapbox/navigation/route/internal/onboard/MapboxOnboardRouterTest.kt:83]: Do we also want to log the `requestId` in `MapboxOnboardRouter#generateErrorMessage` or better wait to refactor that when we migrate to the `HybridRouter`?

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/history/model/HistoryEventPushHistoryRecord.kt [hunk 4]
```diff
@@ -0,0 +1,47 @@
+package com.mapbox.navigation.core.history.model
+
+/**
+ * Allows to read events set with [com.mapbox.navigator.HistoryRecorderHandle.pushHistory]
+ *
+ * @param type corresponded event type of [com.mapbox.navigator.HistoryRecorderHandle.pushHistory]
```
Comment (Guardiola31337) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/history/model/HistoryEventPushHistoryRecord.kt:6]:
```suggestion
 * @param type corresponding to the event type from [com.mapbox.navigator.HistoryRecorderHandle.pushHistory]
```

File: libnavigation-base/src/test/java/com/mapbox/navigation/base/trip/model/roadobject/RoadObjectMapperTest.kt [hunk 5]
```diff
@@ -232,7 +232,8 @@ class RoadObjectMapperTest {
             Date(60),
             Date(80),
             Date(40),
-            null,
+            null, // TODO country codes
+            null, // TODO country codes
```
Comment (Guardiola31337) [inline libnavigation-base/src/test/java/com/mapbox/navigation/base/trip/model/roadobject/RoadObjectMapperTest.kt:236]: cc @korshaknn

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/history/model/HistoryEventPushHistoryRecord.kt [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+ * @param properties corresponded eventJson of [com.mapbox.navigator.HistoryRecorderHandle.pushHistory]
```
Comment (Guardiola31337) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/history/model/HistoryEventPushHistoryRecord.kt:7]:
```suggestion
 * @param properties corresponding to the eventJson from [com.mapbox.navigator.HistoryRecorderHandle.pushHistory]
```

File: libnavigation-router/src/test/java/com/mapbox/navigation/route/internal/onboard/MapboxOnboardRouterTest.kt (hunk 3 above)
Comment (Guardiola31337) [reply libnavigation-router/src/test/java/com/mapbox/navigation/route/internal/onboard/MapboxOnboardRouterTest.kt:83]:
```suggestion
        every { error } returns RouterError(FAILURE_MESSAGE, FAILURE_CODE, REQUEST_ID)
```

File: libnavigation-router/src/test/java/com/mapbox/navigation/route/internal/onboard/MapboxOnboardRouterTest.kt [hunk 7]
```diff
@@ -483,6 +483,7 @@ class MapboxOnboardRouterTest {
             "Error occurred fetching offline route: No suitable edges near location - Code: 171"
         private const val FAILURE_MESSAGE = "No suitable edges near location"
         private const val FAILURE_CODE = 171
+        private const val REQUEST_CODE = 19L
```
Comment (Guardiola31337) [inline libnavigation-router/src/test/java/com/mapbox/navigation/route/internal/onboard/MapboxOnboardRouterTest.kt:486]:
```suggestion
        private const val REQUEST_ID = 19L
```

**Review by Guardiola31337 — APPROVED** [review]
Left some minor comments.

File: libnavigation-router/src/test/java/com/mapbox/navigation/route/internal/onboard/MapboxOnboardRouterTest.kt (hunk 3 above)
Comment (RingerJK) [reply libnavigation-router/src/test/java/com/mapbox/navigation/route/internal/onboard/MapboxOnboardRouterTest.kt:83]: > Do we also want to log the requestId in MapboxOnboardRouter#generateErrorMessage or better wait to refactor that when we migrate to the HybridRouter?

`getRoute` of `Router` is not returning requested, so it no sense to log it. Let's do it when migrating to `RouterInterface`

File: libnavigator/src/main/java/com/mapbox/navigation/navigator/internal/MapboxNativeNavigatorImpl.kt (hunk 2 above)
Comment (RingerJK) [reply libnavigator/src/main/java/com/mapbox/navigation/navigator/internal/MapboxNativeNavigatorImpl.kt:248]: I think it's our next step and `Router` already deprecated, I left todo on top of the class. Also, we don't use `RouterOrigin`, so it's marked as `_`

File: libnavigation-base/src/test/java/com/mapbox/navigation/base/trip/model/roadobject/RoadObjectMapperTest.kt (hunk 5 above)
Comment (korshaknn) [reply libnavigation-base/src/test/java/com/mapbox/navigation/base/trip/model/roadobject/RoadObjectMapperTest.kt:236]: just a reminder, all tests in this class are ignored and can't be fixed right now.
check https://github.com/mapbox/mapbox-navigation-android/issues/4297

**Review by korshaknn — APPROVED** [review]
LGTM

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/history/model/HistoryEventPushHistoryRecord.kt [hunk 8]
```diff
@@ -0,0 +1,47 @@
+package com.mapbox.navigation.core.history.model
+
+/**
+ * Allows to read events set with [com.mapbox.navigator.HistoryRecorderHandle.pushHistory]
```
Comment (LukasPaczos) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/history/model/HistoryEventPushHistoryRecord.kt:4]: Let's not link to a class that we don't expose to developers.

Comment (RingerJK) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/history/model/HistoryEventPushHistoryRecord.kt:4]:
```suggestion
 * Allows to read custom events with **type**-**properties** structure 
```

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/history/model/HistoryEventPushHistoryRecord.kt [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+ *
+ * @param type corresponding to the event type from [com.mapbox.navigator.HistoryRecorderHandle.pushHistory]
+ * @param properties corresponding to the eventJson from [com.mapbox.navigator.HistoryRecorderHandle.pushHistory]
```
Comment (RingerJK) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/history/model/HistoryEventPushHistoryRecord.kt:7]: should docs be added when we start exposing the `class`
```suggestion
 * @param type 
 * @param properties 
```

**Pull request merged**